libraries/standard/mqtt/src/mqtt.c

```diff
@@ -706,12 +754,7 @@ static MQTTStatus_t handleIncomingAck( MQTTContext_t * pContext,
         case MQTT_PACKET_TYPE_UNSUBACK:
             /* Deserialize and give these to the app provided callback. */
             status = MQTT_DeserializeAck( pIncomingPacket, &packetIdentifier, &sessionPresent );
-
-            if( status == MQTTSuccess )
-            {
-                appCallback( pContext, pIncomingPacket, packetIdentifier, NULL );
-            }
-
+            callAppCallback = true;
```

Comment:
In case of UNSUBACK, we always need to call appCallBack regardless of the value of manageKeepAlive? Would you please put a comment explaining why is it so?

Reply:
We always need to call the app callback, except when we receive a PINGRESP and we are managing keep alive. The reason it was initialized to false, is that we need to call the callback before sending any PUBREL or PUBCOMP when we receive a PUBREC/PUBREL, so we call it there instead. The only reason we have this flag is to reduce the complexity of this function (by combining multiple instances of the same block of code into 1), which would otherwise be above our 8 threshold. I will add a comment when callAppCallback is declared